app/services/email_generation_service.rb

```diff
@@ -0,0 +1,56 @@
+class EmailGenerationService
+  LOCK_NAME = "email_generation_worker".freeze
+
+  def self.call(*args)
+    new.call(*args)
+  end
+
+  def call
+    SubscriptionContent.with_advisory_lock(LOCK_NAME, timeout_seconds: 0) do
+      subscription_contents.find_in_batches(batch_size: 500) do |group|
+        to_queue = []
+
+        SubscriptionContent.transaction do
```

Comment:
Is this the right scope for this transaction? Should it be inside the `group.each` so as each email creation and subscription update is in its own?

Reply:
Yeah, I'm using a transaction here for performance rather than consistency. If we process 500 records in one transaction it's faster than processing each one in its own transaction.

Reply:
ok :+1: